crates/orderbook/openapi.yml

```diff
@@ -965,3 +989,18 @@ components:
             the fee after this expiration date. Encoded as ISO 8601 UTC.
           type: string
           example: "1985-03-10T18:35:18.814523Z"
+    SolverCompetitionResponse:
+      type: object
+      description: Maps from solver name to object describing that solver's settlement.
+      additionalProperties:
+        "$ref": "#/components/schemas/SolverSettlement"
+    SolverSettlement:
+      type: object
+      properties:
+        objective_value:
+          type: number
+        gas_estimate:
+          type: number
+        call_data:
```

Comment:
I think this is a very interesting question. Especially solvers with inventory might prefer not to expose their quotes publicly unless they have an upside (winning the batch). I wouldn't want them to not submit a solution if they think their chance of winning is small (but we can also wait and change the model when this happens). 

For users however it would definitely be good to see the alternative prices (to highlight the price improvement competition brings) and maybe even simulate and plot the alternative solution. For a solver it might also be helpful to see the full simulation link to debug why the settlement was failing in the driver.

Maybe we start with all public and then adapt if it becomes an issue (winning solver will reveal their logic for sure)?

Reply:
We can start by only exposing name of winning solver while the competition is on-going. This way, other solvers **don't** have information on:
- Winning objective value (so they can penny the surplus by making the prices slightly better for the users at the expense of the buffers and their weekly payout - as long as its less than 100 COW - in order to win with a slight advantage)
- Calldata - the current routes being proposed and liquidity being used.

Reply:
It feels like our current goal with this route is to make it easier for external solver creators to get insight into the competition. If that is the goal then we don't need to show anything live. It is enough to only make this information available after a winning settlement has been picked and it has been settled.

If the goal is also to provide information that can be used live for example by the frontend to show a user that their order is part of some solutions or has been picked for the currently settling transaction, then we need to include more information.

I feel that there is little reason to show the competition while we're still collecting solutions from all the solvers. This part only takes 10 seconds. Once the driver has all the settlements and has decided which one the winner is but before waiting for the on chain transaction, we can publish all the information. Since the solutions have been collected there shouldn't be worries about pennying.

Reply:
> If the goal is also to provide information that can be used live for example by the frontend to show a user that their order is part of some solutions or has been picked for the currently settling transaction, then we need to include more information.

I think this will eventually be the goal, however I think that we can get around to this once we do it the "right" way (i.e. implement one of your ideas from the discussion https://github.com/cowprotocol/services/issues/175).

Reply:
Created https://github.com/cowprotocol/services/issues/187 to discuss what the "complete" API can look like for use with the FE.